src/Shared/NativeMethodsShared.cs

```diff
@@ -433,57 +433,77 @@ public SystemInformationData()
                 }
                 else
                 {
-                    try
+                    ProcessorArchitectures processorArchitecture = ProcessorArchitectures.Unknown;
+#if !NET35
```

Comment:
Is this too permissive? https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.runtimeinformation.osarchitecture?view=net-5.0 says this is available for .NET Framework 4.71+

Reply:
> Is this too permissive?

C# preprocessor is too limited to express a version greater/less than.

My personal experience is that it works best when using the limiting tfms that code get compiles for because that doesn't require any code changes when the code gets compiled for a new version of the platform.

Reply:
For us this is fine--the only flavor we have that doesn't have this is and will remain net35.

Reply:
(Until we can delete it, which will be a glorious day)